packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx

```diff
@@ -342,6 +355,8 @@ function BipartiteNetworkViz(props: VisualizationProps<Options>) {
     labelTruncationLength: 40,
     emptyNetworkContent,
     getNodeMenuActions,
+    // pass visible node labels
+    visibleNodeLabels: visibleNodeLabels,
```

Comment:
Why not just set `visibleNodeLabels` to be `cleanedData.nodes.flatMap((node) => {
      return { value: node.label, label: node.label };` or similar? Do we need to use state here?

Reply:
@moontrip Thank you for your comment! I may be wrong without further test, but if I remember correctly, it caused some issues of updating visibleNodeLabels correctly, which was due to the fact that it was indirectly related to the Promise of the data request. But it happened with node label control (i.e., for network viz), so in this Bipartite network case, it may just work without using the useState. I will check it out next week if that is the case.

Reply:
@asizemore I checked it carefully and found that there was no issue without using useState as you guessed. 👍 Thus I simplified it by removing unnecessary codes. :)